src/core/server/execution_context/integration_tests/tracing.test.ts

```diff
@@ -0,0 +1,427 @@
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, (context, req, res) =>
+        res.ok({ body: executionContext.get() })
+      );
+
+      await root.start();
+      const response = await kbnTestServer.request
+        .get(root, '/execution-context')
+        .set(new ExecutionContextContainer(parentContext).toHeader())
+        .expect(200);
+
+      expect(response.body).toEqual({ ...parentContext, requestId: expect.any(String) });
+    });
+
+    it('execution context is the same for all the lifecycle events', async () => {
+      const { executionContext, http } = await root.setup();
+      const {
+        createRouter,
+        registerOnPreRouting,
+        registerOnPreAuth,
+        registerAuth,
+        registerOnPostAuth,
+        registerOnPreResponse,
+      } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        return res.ok({ body: executionContext.get()?.toJSON() });
+      });
+
+      let onPreRoutingContext;
+      registerOnPreRouting((request, response, t) => {
+        onPreRoutingContext = executionContext.get()?.toJSON();
+        return t.next();
+      });
+
+      let onPreAuthContext;
+      registerOnPreAuth((request, response, t) => {
+        onPreAuthContext = executionContext.get()?.toJSON();
+        return t.next();
+      });
+
+      let authContext;
+      registerAuth((request, response, t) => {
+        authContext = executionContext.get()?.toJSON();
+        return t.authenticated();
+      });
+
+      let onPostAuthContext;
+      registerOnPostAuth((request, response, t) => {
+        onPostAuthContext = executionContext.get()?.toJSON();
+        return t.next();
+      });
+
+      let onPreResponseContext;
+      registerOnPreResponse((request, response, t) => {
+        onPreResponseContext = executionContext.get()?.toJSON();
+        return t.next();
+      });
+
+      await root.start();
+      const response = await kbnTestServer.request
+        .get(root, '/execution-context')
+        .set(new ExecutionContextContainer(parentContext).toHeader())
+        .expect(200);
+
+      expect(response.body).toEqual({ ...parentContext, requestId: expect.any(String) });
+
+      expect(response.body).toEqual(onPreRoutingContext);
+      expect(response.body).toEqual(onPreAuthContext);
+      expect(response.body).toEqual(authContext);
+      expect(response.body).toEqual(onPostAuthContext);
+      expect(response.body).toEqual(onPreResponseContext);
+    });
+
+    it('propagates context to Elasticsearch scoped client', async () => {
+      const { http } = await root.setup();
+      const { createRouter } = http;
+
+      const router = createRouter('');
+      router.get({ path: '/execution-context', validate: false }, async (context, req, res) => {
+        const { headers } = await context.core.elasticsearch.client.asCurrentUser.ping();
```

Comment:
Could we add an assertion that the ID from `executionContext` here matches the one returned by ES?

Reply:
It's covered by the previous test-suite.